usl_models/usl_models/flood_ml/model.py

```diff
@@ -103,80 +112,27 @@ def _validate_and_preprocess_data(
         data = dataclasses.replace(data, spatiotemporal=st_input)
         return data
 
-    def _model_fit(
-        self, data: FloodModelData, early_stopping: Optional[int]
-    ) -> tf.keras.callbacks.History:
-        """Fits the model on a single batch of FloodModelData.
-
-        Args:
-            data: A FloodModelData object.
-            early_stopping: Optional integer representing the number of epochs to assign
-                as the patience when enabling early stopping. See the train method for
-                more details.
-
-        Returns: A History object containing the training and validation loss
-            and metrics.
-        """
-        inputs = {
-            "spatiotemporal": data.spatiotemporal,
-            "geospatial": data.geospatial,
-            "temporal": data.temporal,
-        }
-        self._model.set_n_predictions(data.storm_duration)
-
-        callbacks = None
-        if early_stopping:
-            es_callback = tf.keras.callbacks.EarlyStopping(
-                monitor="val_loss", mode="min", patience=early_stopping
-            )
-            callbacks = [es_callback]
-
-        history = self._model.fit(
-            inputs,
-            data.labels,
-            batch_size=self._model_params.batch_size,
-            epochs=self._model_params.epochs,
-            validation_split=0.2,
-            callbacks=callbacks,
+    def call(self, input: Input) -> tf.Tensor:
+        return self._model.call(input)
+
+    def call_n(self, input: Input, n: int = 1) -> tf.Tensor:
+        return self._model.call_n(input, n=n)
+
+    def fit(self, dataset: tf.data.Dataset, epochs: int = 1, early_stopping: int | None = None):
+        """Fit the model to the given dataset."""
+        # Create a TensorBoard callback
+        logs = "./logs/" + datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
```

Comment:
what does logging into a 'logs' directory mean in the context of VertexAI training? The logs directory would be in an ephemeral docker container.
This looks like more of a debug option than a default option?

Reply:
Sure, I've moved this callback outside of model.fit.